app/(auth)/sign-in.tsx

```diff
@@ -5,26 +5,78 @@ import { useState } from "react";
 import { ScrollView, Text, View } from "react-native";
 import { SafeAreaView } from "react-native-safe-area-context";
 import { Link, router } from "expo-router";
+import { BASE_URL } from "@/lib/config";
+import AsyncStorage from "@react-native-async-storage/async-storage";
+import AuthError from "@/components/auth/Error";
 
 const SignIn = () => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState("");
   const [form, setForm] = useState({
     email: "",
     password: "",
   });
 
-  const submit = () => {
-    console.log(form.email);
-    console.log(form.password);
-    router.push('/employees')
-  }
+  const submit = async () => {
+    setIsLoading(true);
+
+    try {
+      if (!form.email || !form.password) {
+        setError("All fields are required");
+        return;
+      }
+
+      if (form.password.length < 8) {
+        setError("Password must be at least 8 characters long");
+        return;
+      }
+
+      if (!/^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(form.email)) {
+        setError("Invalid email address");
```

Comment:
Nice work done here!!
Check if there are packages to handle input validation though to make your work easier

Reply:
You don't have to do that on this project. Just check it out

Reply:
Well noted. Thanks